src/pharmpy/modeling/odes.py

```diff
@@ -206,6 +206,84 @@ def set_first_order_elimination(model: Model):
     return model
 
 
+def add_bioavailability_statement(model: Model, add_parameter: bool = True):
+    """Add bioavailability statement for the dose compartment(s) of the model.
+    Can be added as a new parameter or simply as a number.
+
+    Parameters
+    ----------
+    model : Model
+        Pharmpy model
+    add_parameter : bool
+        Add new parameter representing bioavailability or not
+
+    Return
+    ------
+    Model
+        Pharmpy model object
+
+    Examples
+    --------
+    >>> from pharmpy.modeling import *
+    >>> model = load_example_model("pheno")
+    >>> model = add_bioavailability_statement(model)
+
+    See also
+    --------
+    remove_bioavailability
+
+    """
+    odes = model.statements.ode_system
+    if odes is None:
+        raise ValueError(f'Model {model.name} has no ODE system')
+
+    dose_comp = odes.dosing_compartment[0]
+    bio = dose_comp.bioavailability
+
+    if isinstance(bio, sympy.Number):
+        # Bio not defined
+        if add_parameter:
+            model, bio_symb = _add_parameter(model, 'BIO', init=float(bio))
+            ass = Assignment(sympy.Symbol('F'), bio_symb)
+
+            # Set bioavailability of compartment to statement instead
+            cb = CompartmentalSystemBuilder(odes)
+            cb.set_bioavailability(dose_comp, ass.symbol)
+
+            # Add statement to the code
+            model = model.replace(
+                statements=(
+                    model.statements.before_odes
+                    + ass
+                    + CompartmentalSystem(cb)
+                    + model.statements.after_odes
+                )
+            )
+        else:
+            # Add as a number
+            bio_ass = Assignment(sympy.Symbol("BIO"), sympy.Number(1))
+            f_ass = Assignment(sympy.Symbol("F"), bio_ass.symbol)
+
+            cb = CompartmentalSystemBuilder(odes)
+            cb.set_bioavailability(dose_comp, f_ass.symbol)
+
+            model = model.replace(
+                statements=(
+                    bio_ass
+                    + model.statements.before_odes
+                    + f_ass
+                    + CompartmentalSystem(cb)
+                    + model.statements.after_odes
+                )
+            )
+
+    else:
+        # BIO already defined, leave it alone?
+        pass
```

Comment:
I guess it makes sense to not do anything, another option would be to warn or to raise, but we don't do that for e.g. `add_lag_time()`.

Reply:
Nothing changed, but kept else statement in case of future updates to the code